Automattic/wp-calypso
# Developer UX: Add Hosting Configuration item with upsell info

- Closes https://github.com/Automattic/dotcom-forge/issues/1746

## Proposed Changes

* Rename `Developer settings` to `Hosting configuration`
* Take to the top level the thee subitems: `Performance settings`, `Privacy settings` and `Domains and DNS`
* Add upsell information in hosting configuration submenu items and manage plugins.
* Each upsell checks for different site features (Plugins | SFTP)
* Hide Plugins for P2 sites

## Testing Instructions



* Go to `/sites`
* Open the actions menu of a free site
* Observe the info description says `Requires a Business Plan`
* Open the actions menu of a business site
* Observe there is no additional info.

## Screenshots

| **Free** | **Business** |
|---|---|
| <img width="443" alt="Screenshot 2023-03-01 at 17 47 52" src="https://user-images.githubusercontent.com/779993/222221330-a1be3753-5e06-4e07-bd4b-d9822e9698c2.png"> | <img width="379" alt="Screenshot 2023-03-01 at 17 52 58" src="https://user-images.githubusercontent.com/779993/222221897-85997f85-e3af-444d-8ff3-fa1c2eb749b6.png"> |


## Pre-merge Checklist
